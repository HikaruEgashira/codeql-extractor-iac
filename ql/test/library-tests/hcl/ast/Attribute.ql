private import hcl

query predicate attributeExpr(Attribute a) { exists(a.getExpr()) }

query predicate attributeRef(Attribute a, Expr expr) { expr = a.getReference() }
