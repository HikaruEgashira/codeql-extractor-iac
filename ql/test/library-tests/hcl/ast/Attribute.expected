attributeExpr
| sample.hcl:3:5:6:5 | Attribute |
| sample.hcl:9:3:9:31 | Attribute |
| sample.hcl:13:3:13:15 | Attribute |
| sample.hcl:21:3:21:32 | Attribute |
| sample.hcl:22:3:22:22 | Attribute |
| sample.hcl:26:3:26:36 | Attribute |
| sample.hcl:27:3:27:64 | Attribute |
| sample.hcl:28:3:28:60 | Attribute |
| sample.hcl:29:3:29:39 | Attribute |
| sample.hcl:30:3:30:32 | Attribute |
| sample.hcl:31:3:31:38 | Attribute |
attributeRef
| sample.hcl:31:3:31:38 | Attribute | sample.hcl:13:13:13:15 | 1 |
